### PR TITLE
feature: Add firefox to Local grid test runs [NP-1174]

### PR DESCRIPTION
**Pull request by luke-hill**
Same as sister PR for public-website-cucumber

NB: I added parallel functions for the grid tests. I will add parallelisation for the browserstack ones in the next sprint too.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -106,22 +106,18 @@ pipeline {
...
         }
-        stage('Sanity Test') {
+        stage('VRT / Ally Tests') {
             steps {
-                script { env.BUILD_STAGE = 'Sanity Test' }
+                script { env.BUILD_STAGE = 'VRT / Ally Tests' }
                 withDockerSandbox([
                     images['ca-styleguide'],
                     'backstopjs/backstopjs',
                     images['ruby'],
```
Comment (luke-hill) [inline Jenkinsfile:115]: wasn't sure if this was needed for these two tests. So left it in.

File: Jenkinsfile [hunk 2]
```diff
@@ -106,22 +106,18 @@ pipeline {
                 }
             }
         }
-        stage('Sanity Test') {
+        stage('VRT / Ally Tests') {
             steps {
-                script { env.BUILD_STAGE = 'Sanity Test' }
+                script { env.BUILD_STAGE = 'VRT / Ally Tests' }
```
Comment (davidrapson) [inline Jenkinsfile:111]: Use A11y here or the full Accessibility, rather than Ally.

Comment (luke-hill) [reply Jenkinsfile:111]: Should they be grouped together. Or split out into their own stages? Whilst I'm here I can tackle it.

Comment (davidrapson) [reply Jenkinsfile:111]: Granular steps makes sense to me. Don't need to be grouped together

**Comment by luke-hill** [comment]
@simonwgill We get a name conflict on container spin up here: https://jenkins.devops.citizensadvice.org.uk/blue/organizations/jenkins/design-system/detail/PR-482/2/pipeline/144

Thoughts?

**Comment by simonwgill** [comment]
> @simonwgill We get a name conflict on container spin up here: https://jenkins.devops.citizensadvice.org.uk/blue/organizations/jenkins/design-system/detail/PR-482/2/pipeline/144
> 
> Thoughts?

Looks like they're both trying to run on the same agent. There was a thing for that... Pretty sure I was using the label to force it. Which repo did we do again so I can go back and check?

**Comment by luke-hill** [comment]
https://github.com/citizensadvice/public-website-cucumber/blob/master/Jenkinsfile

**Comment by simonwgill** [comment]
Yeah, @luke-hill, using something similar to the withCucumberNode function in public-website-cucumber will definitely help. Obviously you won't need the same crazy env layering in this repo.

File: Jenkinsfile [hunk 3]
```diff
@@ -118,17 +118,34 @@ pipeline {
...
-        stage('VRT / Ally Tests') {
+        stage('Visual Regression Tests') {
             steps {
-                script { env.BUILD_STAGE = 'VRT / Ally Tests' }
+                script { env.BUILD_STAGE = 'Visual Regression Tests' }
                 withDockerSandbox([
                     images['ca-styleguide'],
                     'backstopjs/backstopjs',
-                    images['ruby'],
                     images['wcag'] ]) {
```
Comment (davidrapson) [inline Jenkinsfile:127]: This can go too

**Comment by luke-hill** [comment]
This fails to find the defined method. I have no idea why :( :shrug:

**Comment by luke-hill** [comment]
Being dealt with in 491

**Pull request closed without merging**